src/node_crypto.cc

```diff
@@ -4637,12 +4653,23 @@ void ECDH::SetPublicKey(const FunctionCallbackInfo<Value>& args) {
   EC_POINT* pub = ecdh->BufferToPoint(Buffer::Data(args[0].As<Object>()),
                                       Buffer::Length(args[0].As<Object>()));
   if (pub == nullptr)
-    return;
+    return env->ThrowError("Failed to convert Buffer to EC_POINT");
 
   int r = EC_KEY_set_public_key(ecdh->key_, pub);
   EC_POINT_free(pub);
   if (!r)
-    return env->ThrowError("Failed to convert BN to a private key");
+    return env->ThrowError("Failed to set EC_POINT as the public key");
+}
+
+bool ECDH::IsKeyPairValid() {
+  return 1 == EC_KEY_check_key(key_);
```

Comment:
`EC_KEY_check_key` has side effects, it touches the error stack on error.  You should clear the top of the stack in that case.